brave/brave-core
# Disabling Ads Panel for disallowed regions

Fixes: https://github.com/brave/brave-browser/issues/2562
UI: https://github.com/brave/brave-ui/pull/427

<img width="630" alt="Screen Shot 2019-03-13 at 7 05 04 PM" src="https://user-images.githubusercontent.com/8732757/54326428-960ec380-45c3-11e9-9ab1-318deed2ef15.png">

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on

## Test Plan:
1. Launch Brave with a clean profile, opt in to Rewards
2. If you are inside a supported region (US, CA, DE, FR, GB) you should see Ads enabled and configurable as expected.
3. If not, check that the panel is disabled and cannot be turned on
4. Modify the allowed regions vector here for testing purposes: (Add your own region or remove the one you're in) https://github.com/brave/brave-core/blob/master/vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc#L326
(Unfortunately the flag does not work anymore)
5. Reset profile and rebuild and relaunch Brave.
6. Repeat steps 2 and 3 for a few different regions

## Reviewer Checklist:


cc: @jsecretan
